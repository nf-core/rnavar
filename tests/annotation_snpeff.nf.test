nextflow_pipeline {

    name "Test pipeline"
    script "../main.nf"
    tag "pipeline"
    tag "pipeline_rnavar"

    test("-profile test --tools snpeff --download_cache") {

        when {
            params {
                input = "${projectDir}/tests/csv/1.0/vcf.csv"
                outdir = "$outputDir"
                download_cache = true
                tools = 'snpeff'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // vcf_files: All vcf files
            def vcf_files  = getAllFilesFromDir(params.outdir, include: ['**/*.vcf.gz'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All vcf files
                    vcf_files.collect { file -> [file.getName(), path(file.toString()).vcf.variantsMD5] }
                ).match() }
            )
        }
    }

    test("-profile test --tools snpeff --snpeff_db na") {
        when {
            params {
                input = "${projectDir}/tests/csv/1.0/vcf.csv"
                outdir = "$outputDir"
                snpeff_cache = 's3://annotation-cache/snpeff_cache/'
                snpeff_db = "na"
                tools = 'snpeff'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.failed},
                { assert workflow.stdout.toString().contains("This path is not available within annotation-cache") }
            )
        }
    }
}
