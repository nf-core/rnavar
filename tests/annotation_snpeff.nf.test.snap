{
    "-profile test --tools snpeff --download_cache": {
        "content": [
            11,
            {
                "GATK4_BEDTOINTERVALLIST": {
                    "gatk4": "4.6.1.0"
                },
                "GATK4_INTERVALLISTTOOLS": {
                    "gatk4": "4.6.1.0"
                },
                "GFFREAD": {
                    "gffread": "0.12.7"
                },
                "GTF2BED": {
                    "Rscript": "3.5.1 (2018-07-02)"
                },
                "REMOVE_UNKNOWN_REGIONS": {
                    "python": "3.8.3"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "SNPEFF_DOWNLOAD": {
                    "snpeff": "5.1d"
                },
                "SNPEFF_SNPEFF": {
                    "snpeff": "5.1d"
                },
                "STAR_INDEXVERSION": {
                    "gawk": "5.1.0",
                    "samtools": 1.21,
                    "star": "2.7.11b"
                },
                "TABIX_BGZIPTABIX": {
                    "tabix": 1.2
                }
            },
            [
                "annotation",
                "annotation/test",
                "annotation/test/test_snpEff.ann.vcf.gz",
                "annotation/test/test_snpEff.ann.vcf.gz.tbi",
                "cache",
                "cache/snpeff_cache",
                "cache/snpeff_cache/WBcel235.105",
                "cache/snpeff_cache/WBcel235.105/sequence.I.bin",
                "cache/snpeff_cache/WBcel235.105/sequence.II.bin",
                "cache/snpeff_cache/WBcel235.105/sequence.III.bin",
                "cache/snpeff_cache/WBcel235.105/sequence.IV.bin",
                "cache/snpeff_cache/WBcel235.105/sequence.V.bin",
                "cache/snpeff_cache/WBcel235.105/sequence.X.bin",
                "cache/snpeff_cache/WBcel235.105/sequence.bin",
                "cache/snpeff_cache/WBcel235.105/snpEffectPredictor.bin",
                "cache/versions.yml",
                "pipeline_info",
                "pipeline_info/nf_core_rnavar_software_mqc_versions.yml",
                "reports",
                "reports/multiqc_data",
                "reports/multiqc_data/multiqc.log",
                "reports/multiqc_data/multiqc_citations.txt",
                "reports/multiqc_data/multiqc_data.json",
                "reports/multiqc_data/multiqc_general_stats.txt",
                "reports/multiqc_data/multiqc_snpeff.txt",
                "reports/multiqc_data/multiqc_software_versions.txt",
                "reports/multiqc_data/multiqc_sources.txt",
                "reports/multiqc_data/snpeff_effects.txt",
                "reports/multiqc_data/snpeff_qualities.txt",
                "reports/multiqc_data/snpeff_variant_effects_region.txt",
                "reports/multiqc_plots",
                "reports/multiqc_plots/pdf",
                "reports/multiqc_plots/pdf/general_stats_table.pdf",
                "reports/multiqc_plots/pdf/snpeff_effects-cnt.pdf",
                "reports/multiqc_plots/pdf/snpeff_effects-pct.pdf",
                "reports/multiqc_plots/pdf/snpeff_qualities.pdf",
                "reports/multiqc_plots/pdf/snpeff_variant_effects_region-cnt.pdf",
                "reports/multiqc_plots/pdf/snpeff_variant_effects_region-log.pdf",
                "reports/multiqc_plots/pdf/snpeff_variant_effects_region-pct-log.pdf",
                "reports/multiqc_plots/pdf/snpeff_variant_effects_region-pct.pdf",
                "reports/multiqc_plots/png",
                "reports/multiqc_plots/png/general_stats_table.png",
                "reports/multiqc_plots/png/snpeff_effects-cnt.png",
                "reports/multiqc_plots/png/snpeff_effects-pct.png",
                "reports/multiqc_plots/png/snpeff_qualities.png",
                "reports/multiqc_plots/png/snpeff_variant_effects_region-cnt.png",
                "reports/multiqc_plots/png/snpeff_variant_effects_region-log.png",
                "reports/multiqc_plots/png/snpeff_variant_effects_region-pct-log.png",
                "reports/multiqc_plots/png/snpeff_variant_effects_region-pct.png",
                "reports/multiqc_plots/svg",
                "reports/multiqc_plots/svg/general_stats_table.svg",
                "reports/multiqc_plots/svg/snpeff_effects-cnt.svg",
                "reports/multiqc_plots/svg/snpeff_effects-pct.svg",
                "reports/multiqc_plots/svg/snpeff_qualities.svg",
                "reports/multiqc_plots/svg/snpeff_variant_effects_region-cnt.svg",
                "reports/multiqc_plots/svg/snpeff_variant_effects_region-log.svg",
                "reports/multiqc_plots/svg/snpeff_variant_effects_region-pct-log.svg",
                "reports/multiqc_plots/svg/snpeff_variant_effects_region-pct.svg",
                "reports/multiqc_report.html",
                "reports/snpeff",
                "reports/snpeff/test",
                "reports/snpeff/test/snpEff_summary.html",
                "reports/snpeff/test/test_snpEff.csv",
                "reports/snpeff/test/test_snpEff.genes.txt"
            ],
            [
                "sequence.I.bin:md5,2fd1694bd91cf7952cbad8cfed161e53",
                "sequence.II.bin:md5,bacedbdea89508e108223767fa260a4c",
                "sequence.III.bin:md5,444118a9fb9d0a03c37e86094d8e52a9",
                "sequence.IV.bin:md5,ff756628faa0b71cd65495668c3d82b5",
                "sequence.V.bin:md5,d6ad5476162ac45829f719dd4ee3f4e7",
                "sequence.X.bin:md5,b79bec6cc8f96b8373dac56bab5d0a6c",
                "sequence.bin:md5,ec2bc2ae81755ab90fcf1848bc7ce41f",
                "snpEffectPredictor.bin:md5,1d99251d0405f0a42913ed8b5b2c2fa7",
                "versions.yml:md5,f1b08f4854cb6392d0567334f195e7f6",
                "multiqc_citations.txt:md5,47e39f5f5f05da6bc38d13aa81fe8b6e",
                "multiqc_snpeff.txt:md5,03a2b1c461cb6e5cccac64033a2f6526",
                "snpeff_effects.txt:md5,3c5e9a1c191b77c781dc4d033b1dd1f7",
                "snpeff_qualities.txt:md5,4c059b4e8bf0a64940ad1d6e30efd3a6",
                "snpeff_variant_effects_region.txt:md5,05efd324edadced17ba3cd2b7714af57",
                "test_snpEff.genes.txt:md5,130536bf0237d7f3f746d32aaa32840a"
            ],
            "No BAM files",
            "No CRAM files",
            [
                "test_snpEff.ann.vcf.gz:md5,fd266e623365c7d1837bc9eb088fe70c"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-07T13:20:58.386404456"
    }
}